src/adam_bashforth_methods.jl

```diff
@@ -0,0 +1,647 @@
+            yerr_km1 = h*(g[steporder] - g[steporder-1])*Φ[3,steporder]
+            err_km1 = stepsize_hw92!(h, tdir, y[n], y[n+1], yerr_km1, steporder,
+                            timeout_VS,dof, abstol, reltol, maxstep, norm)[1]
+
+            yerr_k = h*(g[steporder+1] - g[steporder])*Φ[3,steporder+1]
+            err_k = stepsize_hw92!(h, tdir, y[n], y[n+1], yerr_k, steporder,
+                            timeout_VS, dof, abstol, reltol, maxstep, norm)[1]
+
+            # Decrement steporder if LE_{k} and LE_{k-1} < LE_{k+1}
+            if max(err_km1,err_k) <= err_kp1
+                steporder = max(order,steporder - 1)
+
+            elseif consec_const_steps >= steporder && timeout_VO==0 && n>steporder + 3
+                # Given enough consecutive steps taken at a constant step size
+                # we compute truncation error for order implicit k+2
+                Φ_and_Φstar_coefs!(b, Φ, Φstar,dy[n],t,n,steporder+2)
+                Φ_coefs!(Φ, Φstar,dy[n+1],steporder+3)
+
+                yerr_kp2 = h*γ_star(steporder + 3)*Φ[3,steporder+3]
+                err_kp2 = stepsize_hw92!(h,tdir,y[n],y[n+1],yerr_kp2,steporder,
+                            timeout_VS, dof, abstol, reltol, maxstep, norm)[1]
+
+                # Increment steporder if LE_{k+2} < LE_{k+1} and both LE_{k}
+                # LE_{k-1} > LE_{k+1}
+                if err_kp2 < err_kp1
+                    steporder = min(max_order,steporder + 1)
+                    timeout_VO = timeout_VO_const # reset timout_VO
+                end
+            end
+            # Keep timeout_VO 0, or decrement
+            timeout_VO= timeout_VO==0? 0 : timeout_VO-1
+        end # end of adaptive order selection
+
+        #--#####################################################################
+        # Prepare for next step
+        #--#####################################################################
+
+        # Shift data from this step, from the "current step" position to the
+        # "previous step" position for Φstar and Φ
+        Φstar[1,1:min(max(n-1,1),current_steporder+1)] =
+                                Φstar[2,1:min(max(n-1,1),current_steporder+1)]
+        Φ[1,1:min(max(n-1,1),current_steporder+1)] =
+                                Φ[2,1:min(max(n-1,1),current_steporder+1)]
+
+        #=TODO: interpolate tspan points for points = `specified`
+        while tspan[iter]<t[n] + next_h
+            iter+=1
+            h_iter = tspan[iter]-t[n]
+
+            y_iter[iter] = y[n]
+            for j= 1:steporder
+                y_iter[iter] += h_iter*g[j]* Φstar[2,j]
+            end
+        end
+        =#
+
+        # Update stepsize pointers
+        prev_h = h
+        h = next_h
+
+        # Increment our step
+        n= n+1
+    end #while loop over steps
+    return vcat(t), y
+end
+
+###############################################################################
+# Fixed Step Fixed Order (FSFO) Adam-Bashforth Implicit Solver
```

Comment:
Yes. I made `ode_ab_fsfo` mostly for testing purposes. I also implemented the even fixed steps implicit AB method `ode_imp_ab` which is much shorter in length and essentially the same, save for the even stepsizes. Do you think it is worth having `ode_ab_fsfo` in ODE.jl?